cornell-dti/course-plan
# Fix swim test bugs

### Summary 






This pull request fixes #636.

- [x] made swim test requirement a normal requirement
- [x] made swim test unique id a string instead of -1
  - subsequently allowing ap/ib exams and swim test to be affected by overridden fulfillment choices
- [x] distinguished swim test (transfer credit) and beginning swimming (pe course)
- [x] fixed delete transfer credit bug for swim test
- [x] refactored `deleteTransferCredit` to `global-firestore-data`
- [x] changed text in `DeleteCourseModal.vue` (@einc is this fine?)
  - from `This will delete the selected transfer credit from your schedule.`
  - to `This will delete the selected transfer credit.`




### Test Plan 


Delete course modal text should be slightly modified.
![image](https://user-images.githubusercontent.com/34158284/149055508-e46d8243-15c5-42c6-8837-470d01c2ef04.png)

Deleting transfer credit should still work properly.
1. Choose ENG/CS major and add APCS score 5 (or any other AP/IB exam that fulfills a requirement).
2. Go to the requirements sidebar and delete APCS from the slot.
3. Make sure that APCS got deleted. Make sure there's no weird walkthrough bugs.

Swim test requirement should be functional with the bugs fixed.
1. Choose 'no' for `tookSwim` in the onboarding modal.
2. Check that the swim test requirement is in the sidebar and that it's unfulfilled.
3. Add the beginning swimming pe course (either by dragging or manually adding it)
4. Ensure that the swim test requirement is fulfilled.
5. Delete the beginning swimming pe course from your schedule.
6. Choose 'yes' for `tookSwim` in the onboarding modal.
7. Check that the swim test requirement is in the sidebar and that it's fulfilled.
8. Delete the transfer credit via the sidebar and go through the delete course modal.
9. Ensure that the swim test requirement is now unfulfilled.
